src/View/Helper/PaginatorHelper.php

```diff
@@ -524,6 +524,13 @@ public function generateUrlParams(array $options = [], $model = null)
         $paging = $this->params($model);
         $paging += ['page' => null, 'sort' => null, 'direction' => null, 'limit' => null];
 
+        if (!empty($paging['sort']) && !empty($options['sort']) && strpos($options['sort'], '.') === false) {
+            $paging['sort'] = $this->_removeAlias($paging['sort'], $model = null);
```

Comment:
You're assigning to `model` here which creates a side-effect inside a method call.